File: Top_Interview_Questions/387._First_Unique_Character_in_a_String/solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        if not s:
            return -1
        
        # we can use the hash map to store the index of the characters
        # if it is a seen character, then make it a large number
        # and return the minimum index if there is one other than the large numbers
        d = {}
        
        for i,ch in enumerate(s):
            if ch in d:
                d[ch] = float('inf')
            else:
                d[ch] = i
        
        ans = min(d.values())
        if ans == float('inf'):
            return -1
        
        return ans

File: Top_Interview_Questions/387._First_Unique_Character_in_a_String/solution.java
class Solution {
    public int firstUniqChar(String s) {
        int[] count = new int[26];
        for (int i=0; i<s.length(); i++) {
            count[(int)s.charAt(i) - (int)'a']++;
        }

        for (int i=0; i<s.length(); i++) {
            if (count[(int)s.charAt(i) - (int)'a'] == 1)
                return i;
        }

        return -1;
    }
}